### DISCUSSION TITLE
K1C: Timelapse Camera selection doesn't work in Mainsail, but does in Fluidd

### DISCUSSION THREAD

**Discussion by TheWebMachine** [Guilouz/Creality-Helper-Script-Wiki#473] [Issues]
I couldn't figure out why timelapses weren't working. I prefer Mainsail and had Fluidd uninstalled. With cam setup and visible - either manually or using #4 option in tools, same effect - you can't select the camera in Mainsail's Timelapse config and, therefore, timelapses never happen. There are no cameras listed in Mainsail's Timelapse camera dropdown menu (it's disabled) despite showing in Webcams section. 

That said, if I install Fluidd, go to Timelapse settings, see that the option is empty, I can dropdown, select my camera, and timelapses work happily ever after. Mainsail will still show an empty Camera selection box in Timelapse settings despite Fluidd being correct and timelapses working. Mainsail isn't able to see the Webcam entry, whether added manually or via helper script.

I have no clue at all where to begin looking for the cause of this. I suspect something isn't being put into place correctly or as Mainsail expects it to be when running #17 install moonraker timelapse. 

Time for shots!!! 

(screenshots lol)
Mainsail:
![mainsailWebcams](https://github.com/Guilouz/Creality-K1-Series/assets/1045052/750e64d4-8d07-4328-a07d-dcb85b2cd270)

![mainsailTimelapse](https://github.com/Guilouz/Creality-K1-Series/assets/1045052/723fc51a-8fb1-4dc5-8a87-4fc0fd722be8)

Fluidd:
![fluiddTimelapseDropdown](https://github.com/Guilouz/Creality-K1-Series/assets/1045052/1acbd4f3-79db-4fc4-b452-ea2237ed1ebd)
![fluiddTimelapse](https://github.com/Guilouz/Creality-K1-Series/assets/1045052/e3bd8888-f5cd-45d7-b0f8-3991ac37bde3)

Note: This issue exists on both 1.3.3.5 and 1.3.3.8, I just only now got around to posting here about it.

**Comment by TheWebMachine** [comment]
(and, yes, I have the necessary trigger gcode added by my slicer to each layer)

**Comment by TheWebMachine** [comment]
Wait...I stand corrected (yet again)...I just did some reversion testing. I *was* creating timelapses...but the camera selection dialog in Mainsail is definitely broken. It still takes timelapses even without installing Fluidd and selecting the camera. 

Probably just a mainsail bug. Closing this since it isn't your problem *and* it can serve as some context for anyone else who notices the issue.

**Reply by IBSRyaBuzr** [reply]
Where were you finding your Timelapse's? I'm having the same issue, but even SSH in and checking /timelapse, the directory is empty.

**Reply by TheWebMachine** [reply]
At first I couldn't find them, either. Then I found them in Mainsail's Timelapse section.

The interesting thing is, they didn't show up for my past prints before...then I noticed them after I installed Fluidd, so I thought I had been mistaken. Try installing Fluidd and see if your past timelapses show up all of a sudden.
